### ISSUE TITLE
Clash between the mini! enviroment of optidef and the babel package

### ISSUE THREAD

**Issue by chadibou** [jeslago/optidef#11]
First i' like to thank you for this nice package! it really helps when having complicated optimization problems to write. 
However i have a small issue when using the {mini!} environment. When i try and compile, an error message is returned : \begin{mini!} on input line 174 ended by \end{mini!}. \end{mini!}.
The {mini} and {mini*} on the other hand work fine.
I am using the plain { article } class and no special referencing package. Do you have any idea that could help me. Thanks!
Here is a sample of the code: 

```
\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{amsmath,amsfonts,amssymb} 

\begin{document}

\begin{mini!}|l|[2]
	{\substack{\dot{u},\; \dot{\varepsilon}_{e},\; \dot{\varepsilon}_{p},\; \dot{\varepsilon}_{p,int},\\ \dot{\gamma},\; \dot{\gamma}_{int}}}
	{\sum_{q=1}^{N_{Gauss}} f(x)}
	{}{}
	\addConstraint{u_{q}^{(k)}+\dot{u}_{q}}{=\underline{u}_{q}^{(k)}+\underline{\dot{u}}_{q} \qquad} {q \in \Gamma_{\underline{u}}}

\end{document}

```

**Comment by jeslago** [comment]
The example that you provide that does not compile but not because of the optidef package but because you forgot to add the following `\usepackage{optidef}` and `\end{mini!}`. The following code works for me without any issue:

```
\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{amsmath,amsfonts,amssymb} 
\usepackage{optidef}
\begin{document}

\begin{mini!}|l|[2]
	{\substack{\dot{u},\; \dot{\varepsilon}_{e},\; \dot{\varepsilon}_{p},\; \dot{\varepsilon}_{p,int},\\ \dot{\gamma},\; \dot{\gamma}_{int}}}
	{\sum_{q=1}^{N_{Gauss}} f(x)}
	{}{}
	\addConstraint{u_{q}^{(k)}+\dot{u}_{q}}{=\underline{u}_{q}^{(k)}+\underline{\dot{u}}_{q} \qquad}{q \in \Gamma_{\underline{u}}}
\end{mini!}
\end{document}
```

**Comment by chadibou** [comment]
sorry i actually wrote the sample code wrong. I have already included the package and the \end code in my original document but while copying it i missed it since the document is huge^^. 
However, the error persists. The strange this is that it produces a PDF even though it states the error.

**Comment by jeslago** [comment]
Do you get an error with the code I have sent you above? You also had an extra blank space in the \addConstraint command

**Comment by chadibou** [comment]
yes same problem. 
When i just replace mini! by mini it works fine, however i need separate reference for each constraint that's why i wanted to use mini!. Also with mini! and the \breakObjective, the objective function comes with 2 reference number.
here the nearly complicated form of my optimization problem : 
```
\begin{mini!}|l|[2]
	{\substack{\dot{u},\; 
		\dot{\varepsilon}_{e},\; 
		\dot{\varepsilon}_{p},\; 
		\dot{\varepsilon}_{p,int},\\ 
		\dot{\gamma},\; 
		\dot{\gamma}_{int}}
	}
	{\sum_{q=1}^{N_{Gauss}} 
		\frac{1}{2}\dot{\varepsilon}_{e,q}:C_{q}:\dot{\varepsilon}_{e,q} 
		+ \dot{\varepsilon}^{(k)}_{e,q}:C_{q}:\dot{\varepsilon}_{e,q} 
		+ \sqrt{\frac{2}{3}}f_{y}\dot{\gamma}_{q}
		+ \sqrt{\frac{1}{3}}f_{y}\dot{\gamma}_{int,q}
	}{}{}
	\breakObjective{
		-\sum_{q \in \Gamma_{\underline{t}}}
		(\underline{t}_{q}^{(k)}+\underline{\dot{t}}_{q})\dot{u}_{q}
	}
	\addConstraint{(\varepsilon^{(k)}_{e}+\dot{\varepsilon}^{(k)}_{e})+(\varepsilon^{(k)}_{p}+dev(\dot{\varepsilon}^{(k)}_{p})}{=\nabla^{S}(u^{(k)}+\dot{u}^{(k)})\qquad}{in\; \Omega}
	\addConstraint{u_{q}^{(k)}+\dot{u}_{q}}{=\underline{u}_{q}^{(k)}+\underline{\dot{u}}_{q}}{q \in \Gamma_{\underline{u}}}
\end{mini!}
```
here is an image of what i get even though i have the error : 

![image](https://user-images.githubusercontent.com/36533933/36304903-b47a5040-1311-11e8-9b84-d20e6078f76f.png)

**Comment by jeslago** [comment]
For the second reference in the objecetive simply add `\nonumber` at the beginning of the  `\breakObjective` command.

To be honest, I cannot replicate your problem. If I copy your code above, I can compile without any issues. Can you try to compile a document that only includes the optimization problem? The problem might be a class with other command within the large document.

**Comment by chadibou** [comment]
Well anyway thanks a lot. I guess i'll just skip the error since it is still producing a PDF until i find out with which other package it is conflicting.

**Comment by chadibou** [comment]
I guess i found out how you could replicate the problem. Apparently it has something to do with the {babel} package. By commenting the `\usepackage[french,english]{babel}` it works fine. Is there a way to keep them both ?

```
\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{amsmath,amsfonts,amssymb} 
\usepackage{a4wide}
\usepackage[T1]{fontenc}
\usepackage{lmodern}
\usepackage{optidef}
\usepackage[french,english]{babel}


\begin{document}
	
\selectlanguage{english}

\begin{mini!}|l|[2]
	{\substack{\dot{u},\; 
			\dot{\varepsilon}_{e},\; 
			\dot{\varepsilon}_{p},\; 
			\dot{\varepsilon}_{p,int},\\ 
			\dot{\gamma},\; 
			\dot{\gamma}_{int}}
	}
	{\sum_{q=1}^{N_{Gauss}} 
		\frac{1}{2}\dot{\varepsilon}_{e,q}:C_{q}:\dot{\varepsilon}_{e,q} 
		+ \dot{\varepsilon}^{(k)}_{e,q}:C_{q}:\dot{\varepsilon}_{e,q} 
		+ \sqrt{\frac{2}{3}}f_{y}\dot{\gamma}_{q}
		+ \sqrt{\frac{1}{3}}f_{y}\dot{\gamma}_{int,q}
	}{}{}
	\nonumber\breakObjective{
		-\sum_{q \in \Gamma_{\underline{t}}}
		(\underline{t}_{q}^{(k)}+\underline{\dot{t}}_{q})\dot{u}_{q}
	}
	\addConstraint{(\varepsilon^{(k)}_{e}+\dot{\varepsilon}^{(k)}_{e})+(\varepsilon^{(k)}_{p}+dev(\dot{\varepsilon}^{(k)}_{p})}{=\nabla^{S}(u^{(k)}+\dot{u}^{(k)})\qquad}{in\; \Omega}
	\addConstraint{u_{q}^{(k)}+\dot{u}_{q}}{=\underline{u}_{q}^{(k)}+\underline{\dot{u}}_{q}}{q \in \Gamma_{\underline{u}}}
\end{mini!}

\end{document}
```

**Comment by jeslago** [comment]
I found what the error is. For some reason I cannot understand, the naming of the environment clashes with the babel package.

Like if I try to name any environment using a `!` or `:`, e.g. \begin{mini!} or \begin{mini:}, I cannot do it if the babel package is present. However, if I rename any of this commands to simply letters, e.g. \begin{minie}, it works without any issue.

To make it compatible with the babel package, I think I will need to  rename my environments (I have to look into it). But it will take time.

As a provisional patch you can do something very simply for your document. Download the optidef.sty and keep it in the same folder with your main tex file. Open the optidef.sty file and find the following line
```
\DeclareDocumentEnvironment{mini!}
```
and replace it by:
```
\DeclareDocumentEnvironment{minie}
```
Then in your main document do the same, i.e. replace \begin{mini!} by \begin{minie}.

Let me know if that works.

**Comment by chadibou** [comment]
Ok great! it works like a charm. It is apparently just a clash with the name environment.

Thanks man! 👍

**Comment by jeslago** [comment]
You are welcome. Glad that I could help.

**Comment by jeslago** [comment]
This issue has been now solved in commit 86bef453995765bf60880855a1d2f375d0548c18.

A new version of the package will soon be uploaded to CTAN.
